packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts

```diff
@@ -36,18 +37,34 @@ export class StackMetadataBackendOutputStorageStrategy
       new CfnOutput(this.stack, key, { value });
     });
 
-    this.stack.addMetadata(keyName, {
-      version: backendOutputEntry.version,
-      stackOutputs: Object.keys(backendOutputEntry.payload),
-    });
+    const metadata = this.stack.templateOptions.metadata || {};
+    const existingMetadataEntry = metadata[keyName];
+
+    if (existingMetadataEntry) {
+      const excludedStackOutputs = existingMetadataEntry.stackOutputs.filter(
+        (output: string) =>
+          !Object.keys(backendOutputEntry.payload).includes(output)
+      );
+      this.stack.addMetadata(keyName, {
+        version: backendOutputEntry.version,
+        stackOutputs: Object.keys(backendOutputEntry.payload).concat(
+          excludedStackOutputs
+        ),
+      });
```

Comment:
Can this be simplified by using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set ?.

If I understand this. This piece of code is trying to merge two arrays without duplicates right?

Reply:
Done [9843705](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/9843705aeb2cc5ba2894938175d691752ebae988)